lib/prerequisite.js

```diff
@@ -12,41 +12,24 @@ module.exports = (input, pkg, options) => {
 		{
 			title: 'Ping npm registry',
 			skip: () => pkg.private || isExternalRegistry,
-			task: () => pTimeout(
-				(async () => {
-					try {
-						await execa('npm', ['ping']);
-					} catch (_) {
-						throw new Error('Connection to npm registry failed');
-					}
-				})(),
-				15000,
-				'Connection to npm registry timed out'
-			)
+			task: async () => npm.checkConnection()
 		},
 		{
 			title: 'Verify user is authenticated',
 			skip: () => process.env.NODE_ENV === 'test' || pkg.private || isExternalRegistry,
 			task: async () => {
 				let username;
 				try {
-					username = await execa.stdout('npm', ['whoami']);
+					username = await npm.username();
 				} catch (error) {
-					throw new Error(/ENEEDAUTH/.test(error.stderr) ?
+					throw new Error(/ENEEDAUTH/.test(username.stderr) ? // eslint-disable-line unicorn/prefer-type-error
 						'You must be logged in to publish packages. Use `npm login` and try again.' :
 						'Authentication error. Use `npm whoami` to troubleshoot.');
 				}
```

Comment:
This could be a `verifyUserIsAuthenticated()` method.

Reply:
In this case, we do need the username itself though. We could just have `verifyUserIsAuthenticated()` return it.